rotkehlchen/api/rest.py

```diff
@@ -946,6 +946,29 @@ def user_change_password(
         else:
             return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
+    @require_premium_user(active_check=False)
```

Comment:
you don't need this decorator. Even if the user does not have premium he should be able to "delete" whatever entry of rotki premium keys he/she has

Reply:
In the frontend I don't think we have a way of knowing if the current user has invalid keys currently in the system. Basically you would only see the "delete keys" button if the account has premium currently activated (similarly you only see the "replace keys" button in the same manner).

I will try today to regenerate my keys thus invalidating them and see what happens in the frontend.

Reply:
yeah that's how it works. It's same with the eth rpc endpoint. Try to offer the service, if it does not work log it and continue normally.

I would also offer the delete keys in that case too. Same thing which should happen for the eth rpc endpoint.

Reply:
It would just be really weird to show the user a blank form and have a delete button there, no? There is no possible indication that API keys are currently set up (but incorrect).

Reply:
Oh so nothing is shown in the UI then? If so and if they can be easily overwritten then yeah it makes no sense.

Reply:
Yes currently the API does not return API keys (neither for Premium nor for Exchanges), therefore all the user sees is whether a connection is active or not, but cannot see/edit the previously entered keys (but they can replace them by providing valid new ones).